### PR TITLE
PolygonGraphics.hierarchy

### PR DESCRIPTION
**Pull request by mramato**
This can wait until after 1.6 instead of rushing it into master; I'll probably pull it into the `implicit-properties` branch first; to aid with other efforts.

* Added `PolygonHierarchy` to Core to make defining polygons with holes clearer.
* `PolygonGraphics.positions` was deprecated and replaced with `PolygonGraphics.hierarchy`, which evaluates to a `PolygonHierarchy` instead of an array of positions.
* `GeoJsonDataSource` now supports polygons with holes.
* `ConstantProperty` can now hold any value; previously it was limited to values that implemented `equals` and `clones` functions, as well as a few special cases.
* CZML does not yet have support for holes and is still creating an array of positions instead of a `PolygonHierarchy`, which is still supported during the deprecation period.  I'm still trying to determine the best way to do holes in CZML (which will most likely be a separate pull request unless anyone has any objections).
* `PolygonGraphics.positions` created by `GeoJSONDataSource` now evaluate to a `PolygonHierarchy` object instead of an array of positions. __This one is technically a breaking change__ I could probably jump through some additional hoops to avoid this if we think it's important; but it's an easy fix for anyone it may affect.

The overall goal is to move exclusively to the `hierarchy` property for `PolygonGraphics` and have it always evaluate to a `PolygonHierarchy` object.  This simplifies the API for end users.

@bagnell Some of the GeoJSON files we test with look worse when using holes.  I think it may be a big on our end; I'll see if I can pull together some examples for you to look at.

### CODE REVIEW

**Comment by pjcozzi** [comment]
I think this is fine, but I suggest we also keep `positions`, which is consistent with the rest of Cesium, for polygons that do not have any holes.

**Comment by mramato** [comment]
The problem with keeping both is that it makes the API harder to use.  We get away with this for `PolygonGeometry` because it's set via constructor only, you can't ever read the values back. For `PolygonGraphics`, it's set and get.  If we allow for both `positions` and `hierarchy` to exist; then that means anywhere that needs to know the polygon definition needs to check two properties (and also be consistent in how they follow the rules for checking it, i.e. if they are both set use hierarchy instead of positions). What we _can_ do is allow you to set hierarchy to a list of positions as well as an hierarchy (`hierarchy.getValue` would always return a `PolygonHiearchy` instance, but setting could go either way, which is what this branch does now).

I think we need to be careful about making incremental changes here and creating churn.  If you want, I can close this PR and merge this branch into implicit properties and just have them all come in together (when it is ready); this way we can see the big picture rather than trying to make tough api decisions now.

**Comment by mramato** [comment]
@pjcozzi can I get a ruling here; I'd like to get this into master because several branches depend on it.  I'll go a long with whatever you want; we have bigger fish to fry.  Just let me know.

**Comment by pjcozzi** [comment]
I rather not making this renaming decision in haste when we have much bigger things on the radar.  Let's leave it as is.  We can revisit later if needed.

@bagnell can you review this?

File: Source/DataSources/PolygonGeometryUpdater.js [hunk 1]
```diff
@@ -532,12 +540,17 @@ define([
         }
 
         var options = this._options;
-        var positions = Property.getValueOrUndefined(polygon.positions, time, options.polygonHierarchy.positions);
-        if (!defined(positions)) {
+        var hierarchy = Property.getValueOrUndefined(polygon.hierarchy, time, this._scrach);
```
Comment (bagnell) [inline Source/DataSources/PolygonGeometryUpdater.js]: Did you mean to assign to `_scrach` too? Its always `undefined`.

**Comment by bagnell** [comment]
@mramato Just that one comment. The rest looks good.

File: Source/DataSources/PolygonGeometryUpdater.js (hunk 1 above)
Comment (mramato) [inline Source/DataSources/PolygonGeometryUpdater.js]: It's actually not needed at all.  So I deleted the variable and parameter.  Thanks.  This is ready.

**Pull request merged**